### PR TITLE
bpo-9305: Remove east/west references from datetime

### PR DESCRIPTION
**Pull request by slateny**
https://bugs.python.org/issue9305

### CODE REVIEW

File: Doc/library/datetime.rst [hunk 1]
```diff
@@ -1954,8 +1954,8 @@ Examples of working with a :class:`.time` object::
 
 .. method:: tzinfo.utcoffset(dt)
 
-   Return offset of local time from UTC, as a :class:`timedelta` object that is
-   positive east of UTC. If local time is west of UTC, this should be negative.
+   Returns the dfference between local time and UTC as a :class:`timedelta` object.
```
Comment (iritkatriel) [inline Doc/library/datetime.rst]:
```suggestion
   Return the dfference between local time and UTC as a :class:`timedelta` object.
```

See https://www.python.org/dev/peps/pep-0257/#id5 .

Comment (slateny) [reply Doc/library/datetime.rst]: Hmm, in the bpo it was suggested that the nonimperative  `Returns` can be used for documentation, though perhaps not docstring as the PEP is explicit about that. Would you still prefer to change all forms to `Return`?

Comment (arhadthedev) [inline Doc/library/datetime.rst]:
```suggestion
   Returns the difference between local time and UTC as a :class:`timedelta` object.
```

File: Modules/_datetimemodule.c [hunk 2]
```diff
@@ -1213,7 +1213,7 @@ call_utcoffset(PyObject *tzinfo, PyObject *tzinfoarg)
  * doesn't return None or timedelta, TypeError is raised and this
  * returns -1.  If dst() returns an invalid timedelta for a UTC offset,
  * ValueError is raised and this returns -1.  Else *none is set to 0 and
- * the offset is returned (as timedelta, positive east of UTC).
+ * the offset from local time is returned as an int # of minutes.
```
Comment (pganssle) [inline Modules/_datetimemodule.c]: I think better to spell out # as "number"

Comment (pganssle) [inline Modules/_datetimemodule.c]: I think better to spell out # as "number"

File: Doc/library/datetime.rst (hunk 1 above)
Comment (pganssle) [reply Doc/library/datetime.rst]: I personally like `Returns`, but everything else on [`datetime`](https://docs.python.org/3/library/datetime.html) uses `Return`, so let's be consistent with that for now and not let this PR get blocked on the question of what verb form to use.

I'll create a separate issue for switching from `Return` to `Returns`.

Comment (pganssle) [reply Doc/library/datetime.rst:1957]: I personally like `Returns`, but everything else on [`datetime`](https://docs.python.org/3/library/datetime.html) uses `Return`, so let's be consistent with that for now and not let this PR get blocked on the question of what verb form to use.

I'll create a separate issue for switching from `Return` to `Returns`.

Comment (slateny) [reply Doc/library/datetime.rst]: Thanks, I've updated the PR accordingly

**Comment by rhettinger** [comment]
I understand what this PR is trying to do but think it makes users worse off.  For the most part, people understand the UTC offset sign conventions in terms of being east or west of the prime meridian.  While this isn't 100% true, it is a good approximation and mental model.

It would be better to leave the East/West positive/negative references in place and just add a note somewhere in the docs explaining the exceptions to the rule.

**Comment by slateny** [comment]
Hmm, for the docs, what about instead adding an east/west note on the `timedelta` object, something like:

'if the time being compared is east of UTC, `timedelta` is positive. If west of UTC, then `timedelta` is negative'

and keeping the east/west removal under the `offset` function? This way, other functions that refer to `timedelta` (like `dst()`, which doesn't have anything on east/west, but do in the docstring) can all refer to one spot.

Regarding the docstrings, I'm pretty indifferent and would be fine with keeping east/west as-is if it reads better there.

**Review by MaxwellDupre — APPROVED** [review]
Test_datetime produces:
Ran 960 tests in 10.108s
OK (skipped=28)
Also, the Docs for datetime is readable and concise.

**Pull request closed without merging**